mpas_analysis/shared/analysis_task.py

```diff
@@ -89,16 +95,86 @@ def setup_task(config, componentName):  # {{{
     make_directories(plotsDirectory)
 
     if componentName == 'ocean':
+        namelistMap = oceanNamelistMap
         streamMap = oceanStreamMap
         variableMap = oceanVariableMap
     elif componentName == 'seaIce':
+        namelistMap = seaIceNamelistMap
         streamMap = seaIceStreamMap
         variableMap = seaIceVariableMap
     else:
+        namelistMap = None
         streamMap = None
         variableMap = None
 
-    return namelist, runStreams, historyStreams, calendar, streamMap, \
-        variableMap, plotsDirectory  # }}}
+    return namelist, runStreams, historyStreams, calendar, namelistMap, \
+        streamMap, variableMap, plotsDirectory  # }}}
+
+
+def check_analysis_enabled(namelist, analysisOptionName, namelistMap=None,
+                           default=False, raiseException=True):
+    '''
+    Check to make sure a given analysis is turned on, issuing a warning or
+    raising an exception if not.
+
+    Parameters
+    ----------
+    namelist : NameList object
+        for parsing namelist options
+
+    analysisOptionName : str
+        The name of a boolean namelist option indicating whether the given
+        analysis member is enabled
+
+    namelistMap : dict, optional
+        A dictionary of MPAS namelist options that map to their mpas_analysis
+        counterparts.
+
+    default : bool, optional
+        If no analysis option with the given name can be found, indicates
+        whether the given analysis is assumed to be enabled by default.
+
+    raiseException : bool, optional
+        Whether
+
+    Returns
+    -------
+    enabled : bool
+        Whether the given analysis is enabled
+
+    Raises
+    ------
+    RuntimeError
+        If the given analysis option is not found and ``default`` is not
+        ``True`` or if the analysis option is found and is ``False``.  The
+        exception is only raised if ``raiseException = True``.
+
+    Authors
+    -------
+    Xylar Asay-Davis
+
+    Last Modified
+    -------------
+    04/01/2017
+    '''
+
+    try:
+        if namelistMap is None:
+            optionName = analysisOptionName
+        else:
+            optionName = namelist.find_option(namelistMap[analysisOptionName])
+        enabled = namelist.getbool(optionName)
+    except ValueError:
+        enabled = default
+        if default:
+            warnings.warn('WARNING: namelist option {} not found.\n'
+                          'This likely indicates that the simulation you are analyzing was run with an\n'
+                          'older version of MPAS-O that did not support this flag.  Assuming enabled.'.format(analysisOptionName))
```

Comment:
Lately, I've been breaking lines twice, once at close to column 80 and again at the `\n`.  That makes them more readable even if there are multiple code lines per output line.  I can do this here, too.